libraries/ESP32/examples/Serial/onReceiveExample/onReceiveExample.ino

```diff
@@ -70,7 +70,7 @@ SemaphoreHandle_t uart_buffer_Mutex = NULL;
 // task created when onReceive() is used
 void UART0_RX_CB() {
   // take the mutex, waits forever until loop() finishes its processing
-  if (xSemaphoreTake(uart_buffer_Mutex, portMAX_DELAY)) {
+  if (uart_buffer_Mutex != NULL && xSemaphoreTake(uart_buffer_Mutex, portMAX_DELAY)) {
```

Comment:
I think this will fail either way as `xSemaphoreTake` will be called with NULL pointer. I think it's better to lock the sketch if the creation of the mutex fails

Reply:
it will not call `xSemaphoreTake` is `uart_buffer_Mutex` is `NULL`.

Reply:
Yes, my bad. I forgot about the short circuit evaluation.